# CD pak/tre files are uppercase and go in 'assets' subdir
diff a/scremake/PKGBUILD b/scremake/PKGBUILD
--- a/scremake/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/scremake/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,55 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgname=scremake
+pkgver=20261008.r2.gf0380da
+pkgrel=1
+pkgdesc='Strike Commander libRealSpace.'
+arch=(x86_64)
+url=https://remileonard.github.io
+license=(GPL-3.0-only)
+options=(!buildflags)
+depends=(boost-libs glu hicolor-icon-theme libglvnd perl-rename sdl2-compat sdl2_mixer_ext sdl2_ttf xorg-fonts-misc zlib)
+makedepends=(cmake gendesk git gtest icoutils inkscape imagemagick ninja openxr sdl2_mixer_ext sdl2_ttf stb simpleini)
+optdepends=('soundfont-fluid: FluidSynth soundfonts')
+source=("$pkgname"::git+https://github.com/remileonard/libRealSpace.git
+        "$pkgname".patch
+        "$pkgname".sh)
+sha256sums=('SKIP'
+            'c0c5b585e1e3e723172070849c7635a3508178136b807438d92fc10c8956c118'
+            'f52e0593bb26ac90968858791bb80c685947114aae23b3f77e1e058cedb73195')
+
+pkgver() {
+  cd "$pkgname"
+  printf "%s.r%s.g%s" "$(date +%Y%m%d)" "$(git rev-list "$(git rev-list --tags --no-walk --max-count=1)"..HEAD --count)" "$(git rev-parse --short=7 HEAD)"
+}
+
+prepare() {
+  perl -pe 's|(?<=glnch=).*|"/usr/share/scremake/config"|i' "$pkgname".sh > scconfig.sh
+  if [[ ! -f scremake.svg ]]; then
+    icotool -i 8 -x "$pkgname"/resources/strike_commander.ico -o scremake.png
+    inkscape scremake.png --export-plain-svg --export-filename=scremake.svg
+    rm -f scremake.ico scremake.png
+  fi
+  [[ ! -d "build" ]] || rm -rf build
+  patch -Np1 -i "$pkgname".patch
+  perl -pi -e 's/(\.iff)/\U$1/g' "$pkgname"/CMakeLists.txt ; find "$pkgname"/resources -type f -name '*.iff' -exec perl-rename 's|[^/]*\Z|\U$&|' {} +
+}
+
+build() {
+  cmake -S "$pkgname" -B build -G Ninja \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=/usr/share/"$pkgname" \
+    -Wno-policy
+  cmake --build build -j$(nproc)
+}
+
+package() {
+  gendesk -f -n --name='Strike Commander lRS' --comment="${pkgdesc/.}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/$pkgname.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  gendesk -f -n --name='Strike Commander lRS config' --comment="${pkgdesc/.}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/scconfig.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname=scconfig
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname",applications,icons/hicolor/scalable/apps}}
+  DESTDIR="$pkgdir" cmake --build build --target install
+  install -m 644 sc*.desktop "$pkgdir"/usr/share/applications/
+  install -m 755 sc*.sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname".svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/
+  ln -rsf "$pkgdir"/usr/share/"$pkgname"/neosc "$pkgdir"/usr/bin/"$pkgname"
+}
diff a/scremake/scremake.sh b/scremake/scremake.sh
--- a/scremake/scremake.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/scremake/scremake.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,63 @@
+#!/bin/bash
+gmdir="$HOME/games/strike"
+gmcfg="$HOME/.config/scremake"
+gmtgt="assets/VOCLIST.TRE"
+gmprm=""
+glnch="scremake"
+
+gconf() {
+mkdir -p "$gmcfg"
+printf 'gamepath=%s\ngamemod=%s\ngameparm=%s\n' "$gmdir" "$gmtgt" "$gmprm" > "$gmcfg/basedir"
+}
+
+gstart () {
+while IFS= read -r -d '' file; do
+  [[ -d "$gmbdr/$file" && -z "$gm1st" ]] || mkdir -p "$gmbdr/$file"
+done < <(find /usr/share/scremake -mindepth 1 -type d -printf '%P\0')
+while IFS= read -r -d '' file; do
+  [[ -f "$gmbdr/$file" && -z "$gm1st" ]] || cp -f "/usr/share/scremake/$file" "$gmbdr/$file"
+done < <(find /usr/share/scremake -mindepth 2 -type f ! \( -iname '*.ini' \) -printf '%P\0')
+[[ -f "$gmbdr/assets/config.ini" && -z "$gm1st" ]] || cp -f /usr/share/wc3remake/config.ini "$gmbdr/assets/config.ini"
+#[[ -f "/usr/share/soundfonts/FluidR3_GM.sf2" ]] && export SDL_SOUNDFONTS="/usr/share/soundfonts/FluidR3_GM.sf2"
+echo "cd $gmbdr && $glnch $gmpar" | xargs -i -r sh -c "{}"
+}
+
+gtest () {
+gmbse="" gmbdr="" gmgme="" gmgnm="" gmgpm="" gmpar=""
+
+if [[ -f "$gmcfg/basedir" ]]; then
+  while IFS='=' read -r key val || [[ -n "$key" ]]; do
+    case "$key" in
+      gamepath) gmbse="gamepath=$val"; gmbdr="$val" ;;
+      gamemod)  gmgme="gamemod=$val";  gmgnm="$val" ;;
+      gameparm) gmgpm="gameparm=$val"; gmpar="$val" ;;
+    esac
+  done < "$gmcfg/basedir"
+fi
+}
+
+[[ -f "$gmcfg/basedir" ]] || { gconf; gm1st=1; }
+gtest
+if [[ -z "$gmbdr" ]]; then
+  gconf ; gm1st=1
+  gtest
+fi
+if [[ -d "$gmbdr" ]]; then
+  if [[ ! -f "$gmbdr/$gmgnm" ]]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 2 -maxdepth 2 -type f -regextype posix-extended -iregex "$gmbdr/$gmgnm" -printf '%p\n' | grep -Pic "$gmbdr/$gmgnm")"
+    if (( $gmchk == 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be uppercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && find -H "$gmbdr" -depth -exec perl-rename 's|[^/]*\Z|\U$&|' {} +
+    elif (( $gmchk > 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be uppercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    else
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    fi
+  else
+    gstart
+  fi
+else
+  echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  (( $? == 101 )) && gconf
+fi
diff a/scremake/scremake.patch b/scremake/scremake.patch
index e69de29..feb733a 100644
--- a/scremake/scremake.patch	1969-12-31 17:00:00.000000000 -0700
+++ b/scremake/scremake.patch	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,131 @@
+diff a/scremake/CMakeLists.txt b/scremake/CMakeLists.txt
+--- a/scremake/CMakeLists.txt	1969-12-31 17:00:00.000000000 -0700
++++ b/scremake/CMakeLists.txt	1969-12-31 17:00:00.000000000 -0700
+@@ -4,25 +4,120 @@ set(CMAKE_CXX_STANDARD 17)
+ set(CMAKE_CXX_STANDARD_REQUIRED True)
+ add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+ 
+-# Trouver Google Test
+-find_package(GTest CONFIG REQUIRED)
++# Download and include CPM.cmake automatically
++set(CPM_DOWNLOAD_LOCATION "${CMAKE_BINARY_DIR}/cmake/CPM.cmake")
++if(NOT EXISTS "${CPM_DOWNLOAD_LOCATION}")
++    message(STATUS "Downloading CPM.cmake...")
++    file(DOWNLOAD
++        https://github.com/cpm-cmake/CPM.cmake/releases/latest/download/CPM.cmake
++        "${CPM_DOWNLOAD_LOCATION}"
++    )
++endif()
++include("${CPM_DOWNLOAD_LOCATION}")
+ 
+-# Activer les tests
++# System Dependencies
++find_package(GTest REQUIRED)
+ enable_testing()
+ 
+ find_package(SDL2 REQUIRED)
+-find_package(SDL2_mixer_ext CONFIG REQUIRED)
+ find_package(SDL2_ttf CONFIG REQUIRED)
+-find_package(imgui CONFIG REQUIRED)
+ if (APPLE)
+     set(CMAKE_FIND_FRAMEWORK FIRST)
+ endif()
+ set (OpenGL_GL_PREFERENCE LEGACY)
+ find_package(OpenGL REQUIRED)
+ find_package(Boost COMPONENTS system filesystem REQUIRED)
++
++# System SDL2_mixer_ext
++find_package(SDL2_mixer_ext QUIET)
++if(NOT TARGET SDL2_mixer_ext AND NOT TARGET SDL2_mixer_ext::SDL2_mixer_ext)
++    find_library(SDL2_MIXER_EXT_LIB NAMES SDL2_mixer_ext REQUIRED)
++    find_path(SDL2_MIXER_EXT_INCLUDE_DIR NAMES SDL_mixer_ext.h PATH_SUFFIXES SDL2 REQUIRED)
++    add_library(SDL2_mixer_ext UNKNOWN IMPORTED)
++    set_target_properties(SDL2_mixer_ext PROPERTIES
++        IMPORTED_LOCATION "${SDL2_MIXER_EXT_LIB}"
++        INTERFACE_INCLUDE_DIRECTORIES "${SDL2_MIXER_EXT_INCLUDE_DIR}"
++    )
++endif()
++
++if(TARGET SDL2_mixer_ext)
++    if(NOT TARGET SDL2_mixer_ext::SDL2_mixer_ext)
++        add_library(SDL2_mixer_ext::SDL2_mixer_ext ALIAS SDL2_mixer_ext)
++    endif()
++    if(NOT TARGET SDL2_mixer_ext::SDL2_mixer_ext_Static)
++        add_library(SDL2_mixer_ext::SDL2_mixer_ext_Static ALIAS SDL2_mixer_ext)
++    endif()
++endif()
++
++# In-tree ImGui & NFD via CPM
++CPMAddPackage(
++    NAME imgui
++    GITHUB_REPOSITORY ocornut/imgui
++    GIT_TAG v1.92.9
++    GIT_SHALLOW YES
++)
++if(NOT TARGET imgui::imgui)
++    add_library(imgui_static STATIC
++        "${imgui_SOURCE_DIR}/imgui.cpp"
++        "${imgui_SOURCE_DIR}/imgui_draw.cpp"
++        "${imgui_SOURCE_DIR}/imgui_tables.cpp"
++        "${imgui_SOURCE_DIR}/imgui_widgets.cpp"
++        "${imgui_SOURCE_DIR}/backends/imgui_impl_sdl2.cpp"
++        "${imgui_SOURCE_DIR}/backends/imgui_impl_opengl2.cpp"
++    )
++    target_include_directories(imgui_static PUBLIC "${imgui_SOURCE_DIR}" "${imgui_SOURCE_DIR}/backends")
++    target_link_libraries(imgui_static PUBLIC SDL2::SDL2)
++    add_library(imgui::imgui ALIAS imgui_static)
++endif()
++include_directories("${imgui_SOURCE_DIR}" "${imgui_SOURCE_DIR}/backends")
++
++CPMAddPackage(
++    NAME nfd
++    GITHUB_REPOSITORY btzy/nativefiledialog-extended
++    GIT_TAG v1.4.1
++    GIT_SHALLOW YES
++)
++if(TARGET nfd AND NOT TARGET nfd::nfd)
++    add_library(nfd::nfd ALIAS nfd)
++endif()
++# --- SDL2_mixer_ext ---
++find_package(SDL2_mixer_ext QUIET)
++if(NOT TARGET SDL2_mixer_ext AND NOT TARGET SDL2_mixer_ext::SDL2_mixer_ext)
++    find_library(SDL2_MIXER_EXT_LIB NAMES SDL2_mixer_ext)
++    find_path(SDL2_MIXER_EXT_INCLUDE_DIR NAMES SDL_mixer_ext.h PATH_SUFFIXES SDL2)
++    if(SDL2_MIXER_EXT_LIB AND SDL2_MIXER_EXT_INCLUDE_DIR)
++        add_library(SDL2_mixer_ext UNKNOWN IMPORTED)
++        set_target_properties(SDL2_mixer_ext PROPERTIES
++            IMPORTED_LOCATION "${SDL2_MIXER_EXT_LIB}"
++            INTERFACE_INCLUDE_DIRECTORIES "${SDL2_MIXER_EXT_INCLUDE_DIR}"
++        )
++    endif()
++endif()
++
++# --- Header-Only Libraries ---
++find_path(SIMPLEINI_INCLUDE_DIR NAMES SimpleIni.h PATHS /usr/include /usr/local/include)
++if(SIMPLEINI_INCLUDE_DIR)
++    set(SIMPLEINI_INCLUDE_DIRS "${SIMPLEINI_INCLUDE_DIR}" CACHE PATH "SimpleINI include dir" FORCE)
++    include_directories(${SIMPLEINI_INCLUDE_DIRS})
++endif()
++
++find_path(STB_INCLUDE_DIR NAMES stb_image.h PATH_SUFFIXES stb PATHS /usr/include)
++if(STB_INCLUDE_DIR)
++    include_directories(${STB_INCLUDE_DIR})
++endif()
++
++include_directories(/usr/include/SDL2)
++
++# Linker circular static archive fix
++if(UNIX AND NOT APPLE)
++    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--start-group")
++    link_libraries(-Wl,--start-group)
++    link_libraries(-Wl,--end-group)
++endif()
++
+ add_subdirectory(src/executables/sc)
+-add_subdirectory(src/executables/to)
+-add_subdirectory(src/executables/debugger)
++#add_subdirectory(src/executables/to)
++#add_subdirectory(src/executables/debugger)
+ add_subdirectory(src/executables/config)
+ if (WIN32)
+     add_subdirectory(src/executables/vrstrike)
